src/cata_tiles.cpp

```diff
@@ -972,51 +962,156 @@ void cata_tiles::draw_rhombus(int destx, int desty, int size, SDL_Color color, i
     }
 }
 
+//converts a submap x/y location to a string with the contents "x_y"
+//should be unique enough to track submaps ingame
+static std::string get_submap_id(const tripoint &t)
+{
+    std::stringstream ss;
+    ss << (t.x / SEEX) << "_" << (t.y / SEEY);
+    return ss.str();
+}
+
+//creates the texture that individual minimap updates are drawn to
+//later, the main texture is drawn to the display buffer
+//the surface is needed to determine the color format needed by the texture
+SDL_Texture_Ptr cata_tiles::create_minimap_cache_texture(int tile_width, int tile_height)
+{
+    SDL_Surface_Ptr temp = create_tile_surface();
+    SDL_Texture_Ptr tex ( SDL_CreateTexture(renderer, temp->format->format, SDL_TEXTUREACCESS_TARGET,
+                                            tile_width, tile_height) );
+    return tex;
+}
+
+//resets the touched and drawn properties of each active submap cache
+void cata_tiles::prepare_minimap_cache_for_updates()
+{
+    for(auto &mcp : minimap_cache) {
+        mcp.second->touched = false;
+        mcp.second->drawn = false;
+    }
+}
+
+//deletes the mapping of unused submap caches from the main map
+//the touched flag prevents deletion
+void cata_tiles::clear_unused_minimap_cache()
+{
+    std::vector<std::string> unused_ids;
+    for(auto &mcp : minimap_cache) {
+        if(!mcp.second->touched) {
+            unused_ids.push_back(mcp.first);
+        }
+    }
+    for(int i = 0; i < static_cast<int>(unused_ids.size()); i++) {
+        minimap_cache.erase(unused_ids[i]);
+    }
```

Comment:
For some reason, I was thinking erase invalidated the iterator like how vector does.